src/core.c/CompUnit/Repository/Installation.pm6

```diff
@@ -23,16 +23,33 @@ class CompUnit::Repository::Installation does CompUnit::Repository::Locally does
         }
     }
 
+    method !prefix-writeable {
+        if Rakudo::Internals.IS-WIN {
```

Comment:
I think it might be ok to cache the result of the windows condition. It adds an edge case when the repo e.g. becomes unwritable when it was writable moments earlier, but it potentially saves enough file creation attempts in the 99%+ cases to be worth it.